tutorials/community/reinforcement_learning/pipeline_reinforcement_learning_vertex_ai/pipeline_reinforcement_learning_vertex_ai.ipynb

```diff
@@ -0,0 +1,1780 @@
+      },
+      "outputs": [],
+      "source": [
+        "# Dataset parameters\n",
+        "RAW_DATA_PATH = f\"{BUCKET_NAME}/[your-dataset-dir]/u.data\"  # @param {type:\"string\"} Location of the MovieLens 100K dataset's \"u.data\" file.\n",
+        "\n",
+        "# BigQuery parameters (used for the Generator, Ingester, Logger)\n",
+        "BIGQUERY_DATASET_ID = f\"{PROJECT_ID}.movielens_dataset\"  # @param {type:\"string\"} BigQuery dataset ID as `project_id.dataset_id`.\n",
+        "BIGQUERY_LOCATION = \"us\"  # @param {type:\"string\"} BigQuery dataset region.\n",
+        "BIGQUERY_TABLE_ID = f\"{BIGQUERY_DATASET_ID}.training_dataset\"  # @param {type:\"string\"} BigQuery table ID as `project_id.dataset_id.table_id`."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "fuN1uU27VaU3"
+      },
+      "source": [
+        "#### Set additional configurations\n",
+        "\n",
+        "You may use the default values below as is."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "H3530hdGGilo"
+      },
+      "outputs": [],
+      "source": [
+        "# Pipeline parameters\n",
+        "PIPELINE_NAME = \"movielens-pipeline\"  # Pipeline display name.\n",
+        "ENABLE_CACHING = False  # Whether to enable execution caching for the pipeline.\n",
+        "PIPELINE_ROOT = f\"{BUCKET_NAME}/pipeline\"  # Root directory for pipeline artifacts.\n",
+        "PIPELINE_SPEC_PATH = \"metadata_pipeline.json\"  # Path to pipeline specification file.\n",
+        "OUTPUT_COMPONENT_SPEC = \"output-component.yaml\"  # Output component specification file.\n",
+        "\n",
+        "# BigQuery parameters (used for the Generator, Ingester, Logger)\n",
+        "BIGQUERY_TMP_FILE = \"tmp.json\"  # Temporary file for storing data to be loaded into BigQuery.\n",
+        "BIGQUERY_MAX_ROWS = 5  # Maximum number of rows of data in BigQuery to ingest.\n",
+        "\n",
+        "# Dataset parameters\n",
+        "TFRECORD_FILE = f\"{BUCKET_NAME}/trainer_input_path/*\"  # TFRecord file to be used for training.\n",
+        "\n",
+        "# Logger parameters (also used for the Logger hook in the prediction container)\n",
+        "LOGGER_PUBSUB_TOPIC = \"logger-pubsub-topic\"  # Pub/Sub topic name for the Logger.\n",
+        "LOGGER_CLOUD_FUNCTION = \"logger-cloud-function\"  # Cloud Functions name for the Logger."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "iDrQfmBtAkpC"
+      },
+      "source": [
+        "#### Download dataset from `gs://cloud-samples-data/vertex-ai/growth-rl-demo/u.data`"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "0YwZd9-4AyCK"
+      },
+      "outputs": [],
+      "source": [
+        "! gsutil cp gs://cloud-samples-data/vertex-ai/growth-rl-demo/u.data $RAW_DATA_PATH"
```

Comment:
Why do we need to copy the dataset? Why not use it form the original location?

Reply:
We don't expect users to download the dataset themselves, although they can. As per [this notebook template](https://github.com/GoogleCloudPlatform/ai-platform-samples/blob/master/ai-platform-unified/notebooks/notebook_template.ipynb) and the instruction "Googlers: Add your dataset to the public samples bucket within gs://cloud-samples-data/ai-platform-unified, if it doesn't already exist there.}", we wanted to host the dataset in a public bucket and add this command to download the dataset for users.

Reply:
It's good that the dataset is public.

What I meant was why should the user copy this public dataset from `gs://cloud-samples-data` to their own bucket?

P.S. The copy can be needed if the BigQuery is incompatible with the `gs://cloud-samples-data` bucket. But in that case it would be great to create/find a public bucket that is compatible.

Reply:
This is a great point. A MovieLens environment from TF-Agents will read the public dataset, not BigQuery. But the environment can indeed load form the public bucket directly. I will make changes to the notebooks.